test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,226 @@
+{
+    for (int32_t length : {50, 5000, 50'000})
+        b->Args({length, 5});
+
+    b->Args({500, 1'000});
+}
+
+struct fm_index_seqan2;
+struct fm_index_seqan3;
+struct bi_fm_index_seqan2;
+struct bi_fm_index_seqan3;
+
+template <typename index_tag_t, typename rng_t>
+class sequence_generator
+{
+private:
+    size_t length{50'000};
+    size_t collection_size{5};
+
+    using alphabet_t = seqan3::innermost_value_type_t<rng_t>;
+    using inner_rng_t = std::conditional_t<seqan3::dimension_v<rng_t> == 1,
+                                           rng_t,
+                                           std::ranges::range_value_t<rng_t>>;
+    using fm_index_seqan3_t = decltype(seqan3::fm_index{std::declval<rng_t>()});
+    using bi_fm_index_seqan3_t = decltype(seqan3::bi_fm_index{std::declval<rng_t>()});
+
+    inner_rng_t make_single_seqan3(size_t seed = 0)
+    {
+        // The characters with rank 254 and 255 are reserved for the indices.
+        if constexpr (seqan3::alphabet_size<alphabet_t> > 254)
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed)
+                   | seqan3::views::persist
+                   | seqan3::views::to_rank
+                   | std::views::transform([] (uint8_t const r) { return std::min<uint8_t>(r, 253); })
+                   | seqan3::views::rank_to<alphabet_t>
+                   | seqan3::views::to<inner_rng_t>;
+        }
+        else
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed) | seqan3::views::to<inner_rng_t>;
+        }
+    }
+
+    rng_t make_collection_seqan3()
+    {
+        rng_t result;
+        for (size_t i = 0; i < collection_size; ++i)
+            result.emplace_back(make_single_seqan3(i));
+
+        return result;
+    }
+
+    rng_t make_sequence_seqan3()
+    {
+        if constexpr(seqan3::dimension_v<rng_t> == 1)
+            return make_single_seqan3();
+        else
+            return make_collection_seqan3();
+    }
+
+#if SEQAN3_HAS_SEQAN2
+    // Map the SeqAn3 alphabet to its SeqAn2 equivalent.
+    using seqan2_alphabet_t = std::conditional_t<std::same_as<alphabet_t, seqan3::dna4>, seqan::Dna,
+                                std::conditional_t<std::same_as<alphabet_t, seqan3::aa27>, seqan::AminoAcid, char>>;
+    // SeqAn2: `seqan::String` and `seqan::StringSet`, SeqAn3: `std::vector` and `std::vector<std::vector>`.
+    using seqan2_rng_t = std::conditional_t<seqan3::dimension_v<rng_t> == 1,
+                                            seqan::String<seqan2_alphabet_t>,
+                                            seqan::StringSet<seqan::String<seqan2_alphabet_t>>>;
+    using index_cfg = seqan::FastFMIndexConfig<void, uint32_t>;
+    using fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::FMIndex<void, index_cfg>>;
+    using bi_fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>>;
+
+    seqan2_rng_t make_sequence_seqan2()
```

Comment:
We should introduce a `seqan3::test::generate_seqan2_sequence` into the global test include header since this is something we will need more often.

Reply:
https://github.com/seqan/seqan3/blob/1c2b720f52d033924167949932fe231e24fd9db3/test/include/seqan3/test/performance/sequence_generator.hpp#L77-L92

Reply:
or maybe templatize over the range_type?